path_credentials.go

```diff
@@ -39,115 +38,104 @@ func (b *tfBackend) terraformToken() *framework.Secret {
 				Description: "Terraform Token",
 			},
 		},
-		Renew:  b.terraformTokenRenew,
 		Revoke: b.terraformTokenRevoke,
+		Renew:  b.terraformTokenRenew,
 	}
 }
 
 func (b *tfBackend) pathCredentialsRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	roleName := d.Get("name").(string)
 
-	role, err := b.credentialRead(ctx, req.Storage, roleName)
+	roleEntry, err := b.getRole(ctx, req.Storage, roleName)
 	if err != nil {
-		return nil, errwrap.Wrapf("error retrieving role: {{err}}", err)
+		return nil, fmt.Errorf("error retrieving role: %w", err)
 	}
 
-	if role == nil {
+	if roleEntry == nil {
 		return nil, errors.New("error retrieving role: role is nil")
 	}
 
-	return b.createToken(ctx, req.Storage, roleName, role)
-}
-
-func (b *tfBackend) createToken(ctx context.Context, s logical.Storage, roleName string, roleEntry *terraformRoleEntry) (*logical.Response, error) {
-	client, err := b.getClient(ctx, s)
-	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+	if roleEntry.UserID != "" {
+		return b.createUserCreds(ctx, req, roleEntry)
 	}
 
-	walID, err := framework.PutWAL(ctx, s, terraformTokenType, &walEntry{
```

Comment:
Is the WAL logic no longer needed?

Reply:
Correct

Reply:
+1, I saw that the WAL logic was removed from backend.goL55-56 and was curious about this change.